twisted/conch/insults/insults.py

```diff
@@ -689,196 +694,200 @@ def tilde(self, proto, handler, buf):
-        self.write('\x1b7')
+        self.write(b'\x1b7')
 
 
     def restoreCursor(self):
         self.cursorPos = self._savedCursorPos
         del self._savedCursorPos
-        self.write('\x1b8')
+        self.write(b'\x1b8')
 
 
     def setModes(self, modes):
         # XXX Support ANSI-Compatible private modes
-        self.write('\x1b[%sh' % (';'.join(map(str, modes)),))
+        self.write(b'\x1b[%sh' %
+                   (b';'.join([intToBytes(mode) for mode in modes])))
 
 
     def setPrivateModes(self, modes):
-        self.write('\x1b[?%sh' % (';'.join(map(str, modes)),))
+        self.write(b'\x1b[?%sh' %
+                   (b';'.join([intToBytes(mode) for mode in modes])))
 
 
     def resetModes(self, modes):
         # XXX Support ANSI-Compatible private modes
-        self.write('\x1b[%sl' % (';'.join(map(str, modes)),))
+        self.write(b'\x1b[%sl' %
+                   (b';'.join([intToBytes(mode) for mode in modes])))
 
 
     def resetPrivateModes(self, modes):
-        self.write('\x1b[?%sl' % (';'.join(map(str, modes)),))
+        self.write(b'\x1b[?%sl' %
+                   (b';'.join([intToBytes(mode) for mode in modes])))
 
 
     def applicationKeypadMode(self):
-        self.write('\x1b=')
+        self.write(b'\x1b=')
 
 
     def numericKeypadMode(self):
-        self.write('\x1b>')
+        self.write(b'\x1b>')
 
 
     def selectCharacterSet(self, charSet, which):
         # XXX Rewrite these as dict lookups
         if which == G0:
-            which = '('
+            which = b'('
         elif which == G1:
-            which = ')'
+            which = b')'
         else:
             raise ValueError("`which' argument to selectCharacterSet must be G0 or G1")
         if charSet == CS_UK:
-            charSet = 'A'
+            charSet = b'A'
         elif charSet == CS_US:
-            charSet = 'B'
+            charSet = b'B'
         elif charSet == CS_DRAWING:
-            charSet = '0'
+            charSet = b'0'
         elif charSet == CS_ALTERNATE:
-            charSet = '1'
+            charSet = b'1'
         elif charSet == CS_ALTERNATE_SPECIAL:
-            charSet = '2'
+            charSet = b'2'
         else:
             raise ValueError("Invalid `charSet' argument to selectCharacterSet")
-        self.write('\x1b' + which + charSet)
+        self.write(b'\x1b' + which + charSet)
 
 
     def shiftIn(self):
-        self.write('\x15')
+        self.write(b'\x15')
 
 
     def shiftOut(self):
-        self.write('\x14')
+        self.write(b'\x14')
 
 
     def singleShift2(self):
-        self.write('\x1bN')
+        self.write(b'\x1bN')
 
 
     def singleShift3(self):
-        self.write('\x1bO')
+        self.write(b'\x1bO')
 
 
     def selectGraphicRendition(self, *attributes):
         attrs = []
         for a in attributes:
             attrs.append(a)
-        self.write('\x1b[%sm' % (';'.join(attrs),))
+        self.write(b'\x1b[%sm' % (b';'.join(attrs),))
```

Comment:
PEP 461 only works in Python 3.5 and higher.
Twisted supports Python 2.7, 3.3, 3.4, 3.5 and higher.